### ISSUE TITLE
How can I count the number of label class in my training dataset?

### ISSUE THREAD

**Issue by Ala1412** [ultralytics/yolov5#7793]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I am working on yolov5 with my custom dataset. After the training, it only shows the count of label class of the validation dataset. How can I get the counting of label class in my training dataset?

**Comment by glenn-jocher** [comment]
@Ala1412 you can run val.py on your train set with `python val.py --task train`

You can also look at labels.jpg to see a histogram of training set class counts:
![image](https://user-images.githubusercontent.com/26833433/168258991-c8943abd-24ea-4438-816d-fda9a8a02e12.png)

**Comment by Ala1412** [comment]
> @Ala1412 you can run val.py on your train set with `python val.py --task train`
> 
> You can also look at labels.jpg to see a histogram of training set class counts: ![image](https://user-images.githubusercontent.com/26833433/168258991-c8943abd-24ea-4438-816d-fda9a8a02e12.png)

Thanks for the reply, but I want to know if the "train" is a folder or what? Is it contain both the training images and txt file?

**Comment by glenn-jocher** [comment]
@Ala1412 👋 Hello! Thanks for asking about YOLOv5 🚀 **dataset formatting**. To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths), 2) the number of classes `nc` and 3) a list of class `names`:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
nc: 80  # number of classes
names: [ 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush' ]  # class names
```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>


Good luck 🍀 and let us know if you have any other questions!

**Comment by renzodamgo** [comment]
Is there a way to get the actual label numbers, not only the bar plot?

**Comment by glenn-jocher** [comment]
@renzodamgo yes, val.py prints these analytics. To run val.py on a train set:
```
python val.py --task train --data DATA.yaml
```

**Comment by renzodamgo** [comment]
I have a custom dataset so i need to put my weights too:
```
!python val.py --task train --weights ./runs/train/exp9/weights/best.pt --data facemask.yaml 
```
![Captura de Pantalla 2022-05-25 a la(s) 6 14 17 p  m](https://user-images.githubusercontent.com/23220342/170384355-4dcd70d6-627c-429c-b36d-c92c883dd7c9.png)

And It worked! 🥳  Thank you @glenn-jocher

**Comment by renzodamgo** [comment]
Sorry to bother again but is there a way to get this data from every label from the validation dataset?
- True Positive(TP)
- False Positive(FP)
- False Negative(FN)

**Comment by glenn-jocher** [comment]
@renzodamgo val.py outputs tp and fp here:
https://github.com/ultralytics/yolov5/blob/1dcb77499869d64e5dea7c5bd38357c92f5eff29/val.py#L266
